src/iterators.jl

```diff
@@ -341,10 +341,10 @@ end
     `InterfaceIterator` is stateful and should not be used for things other than `for`-looping
     (e.g. broadcasting over, or collecting the iterator may yield unexpected results).
 """
-struct InterfaceIterator{IC <: InterfaceCache, G <: Grid}
+struct InterfaceIterator{IC <: InterfaceCache, G <: AbstractGrid, Topology <: AbstractTopology}
     cache::IC
     grid::G
-    topology::ExclusiveTopology
+    topology::Topology
```

Comment:
```suggestion
    topology::TopologyType
```

Or smth in this direction. I think using a name which could be some actual type name can bite us in the future.